File: haskell/rna-transcription/DNA.hs
module DNA (toRNA) where

-- | toRNA returns the RNA complement for a given DNA strand.
toRNA :: String -> String
toRNA = map f
  where f 'G' = 'C'
        f 'C' = 'G'
        f 'T' = 'A'
        f 'A' = 'U'
        f _   = undefined

File: python/rna-transcription/dna.py
from string import maketrans


class DNA(object):
    """Represent a strand of DNA."""

    __rna_translation_table = maketrans('GCTA', 'CGAU')

    def __init__(self, dna_string):
        self.dna_string = dna_string

    def to_rna(self):
        """Return the RNA complement of the DNA strand."""
        return self.dna_string.translate(self.__rna_translation_table)
